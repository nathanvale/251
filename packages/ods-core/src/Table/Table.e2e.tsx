/* eslint-disable react/display-name */
import { E2ETests } from "@origin-digital/ods-types";
import { generateTable } from "./Table.helper";

export const tests: E2ETests = [
  {
    label: "Default Table",
    Code: () => generateTable(),
    responsive: true,
  },
  {
    label: "Dense table",
    Code: () => generateTable({ size: "small" }),
    responsive: true,
  },
  {
    label: "Small text size",
    Code: () => generateTable({ textVariant: "body-small" }),
    responsive: true,
  },
  {
    label: "Striped rows",
    Code: () => generateTable({ striped: true }),
    responsive: true,
  },
  {
    label: "Bordered",
    Code: () => generateTable({ bordered: true }),
    responsive: true,
  },
  {
    label: "Hover effect",
    Code: () => generateTable({ hover: true }),
    responsive: true,
  },
  {
    label: "Sticky header",
    Code: () => generateTable({ stickyHeader: true }),
    responsive: true,
  },
  {
    label: "Without header",
    Code: () => generateTable({}, { hideTableHead: true }),
    responsive: true,
  },
];
