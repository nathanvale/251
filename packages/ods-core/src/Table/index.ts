export { Table, defaultProps, TableContext } from "./Table";
export { generateTable } from "./Table.helper";
export type TableProps = import("./Table").TableProps;
