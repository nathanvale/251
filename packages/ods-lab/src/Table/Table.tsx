import React from "react";
import MuiTable, {TableProps as ITable} from "@material-ui/core/Table";

export interface TableProps extends ITable {
  "data-id": string;
  dense?: boolean;
}

export const Table = (props: TableProps) => (
  <MuiTable {...props} size={props.dense ? "small" : props.size}>
    {props.children}
  </MuiTable>
);
