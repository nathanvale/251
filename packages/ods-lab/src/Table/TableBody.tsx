import React from "react";
import MuiTableBody, {
  TableBodyProps as ITableBody,
} from "@material-ui/core/TableBody";

export interface TableBodyProps extends ITableBody {
  "data-id": string;
}

export const TableBody = (props: TableBodyProps) => (
  <MuiTableBody {...props}>{props.children}</MuiTableBody>
);
