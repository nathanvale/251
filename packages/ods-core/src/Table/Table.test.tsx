import { render } from "@origin-digital/ods-testing-library";
import { generateTable } from "./Table.helper";
import { TableProps } from "..";

const renderTable = (tableProps: Partial<TableProps> = {}) =>
  render(generateTable(tableProps));

test("It renders", () => {
  const { container } = renderTable();
  expect(container).toMatchSnapshot();
});

test("It uses small padding", () => {
  const { container } = renderTable({ size: "small" });
  expect(container).toMatchSnapshot();
});

test("It uses medium padding", () => {
  const { container } = renderTable({ size: "medium" });
  expect(container).toMatchSnapshot();
});

test("It adds borders", () => {
  const { container } = renderTable({ bordered: true });
  expect(container).toMatchSnapshot();
});

test("It adds stripes", () => {
  const { container } = renderTable({ striped: true });
  expect(container).toMatchSnapshot();
});

test("It adds hover", () => {
  const { container } = renderTable({ hover: true });
  expect(container).toMatchSnapshot();
});

test("It adds sticky header", () => {
  const { container } = renderTable({ stickyHeader: true });
  expect(container).toMatchSnapshot();
});
