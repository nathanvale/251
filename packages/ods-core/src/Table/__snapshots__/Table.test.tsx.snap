// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It adds borders 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root Table-withBorder-184"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-191"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-194 TableCell-withPadding-199 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-198 TableCell-withFontSize-196 TableCell-withFontSize-200 TableCell-withBorder-197"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-194 TableCell-withPadding-214 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-213 TableCell-withFontSize-196 TableCell-withFontSize-215 TableCell-withBorder-197"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-194 TableCell-withPadding-217 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-216 TableCell-withFontSize-196 TableCell-withFontSize-218 TableCell-withBorder-197"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-194 TableCell-withPadding-228 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-227 TableCell-withFontSize-196 TableCell-withFontSize-229 TableCell-withBorder-197"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-194 TableCell-withPadding-231 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-230 TableCell-withFontSize-196 TableCell-withFontSize-232 TableCell-withBorder-197"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-194 TableCell-withPadding-234 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-233 TableCell-withFontSize-196 TableCell-withFontSize-235 TableCell-withBorder-197"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-194 TableCell-withPadding-237 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-236 TableCell-withFontSize-196 TableCell-withFontSize-238 TableCell-withBorder-197"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-194 TableCell-withPadding-240 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-239 TableCell-withFontSize-196 TableCell-withFontSize-241 TableCell-withBorder-197"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-194 TableCell-withPadding-243 TableCell-withVerticalAlign-193 TableCell-withVerticalAlign-242 TableCell-withFontSize-196 TableCell-withFontSize-244 TableCell-withBorder-197"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It adds class name 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root myCustomClass"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-435"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-438 TableCell-withPadding-443 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-442 TableCell-withFontSize-440 TableCell-withFontSize-444 TableCell-withoutFirstLastPadding-439"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-438 TableCell-withPadding-458 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-457 TableCell-withFontSize-440 TableCell-withFontSize-459 TableCell-withoutFirstLastPadding-439"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-438 TableCell-withPadding-461 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-460 TableCell-withFontSize-440 TableCell-withFontSize-462 TableCell-withoutFirstLastPadding-439"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-438 TableCell-withPadding-472 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-471 TableCell-withFontSize-440 TableCell-withFontSize-473 TableCell-withoutFirstLastPadding-439"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-438 TableCell-withPadding-475 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-474 TableCell-withFontSize-440 TableCell-withFontSize-476 TableCell-withoutFirstLastPadding-439"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-438 TableCell-withPadding-478 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-477 TableCell-withFontSize-440 TableCell-withFontSize-479 TableCell-withoutFirstLastPadding-439"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-438 TableCell-withPadding-481 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-480 TableCell-withFontSize-440 TableCell-withFontSize-482 TableCell-withoutFirstLastPadding-439"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-438 TableCell-withPadding-484 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-483 TableCell-withFontSize-440 TableCell-withFontSize-485 TableCell-withoutFirstLastPadding-439"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-438 TableCell-withPadding-487 TableCell-withVerticalAlign-437 TableCell-withVerticalAlign-486 TableCell-withFontSize-440 TableCell-withFontSize-488 TableCell-withoutFirstLastPadding-439"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It adds hover 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-313"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-316 TableCell-withPadding-321 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-320 TableCell-withFontSize-318 TableCell-withFontSize-322"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-316 TableCell-withPadding-336 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-335 TableCell-withFontSize-318 TableCell-withFontSize-337"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-316 TableCell-withPadding-339 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-338 TableCell-withFontSize-318 TableCell-withFontSize-340"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root TableRow-withHover-343"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-316 TableCell-withPadding-350 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-349 TableCell-withFontSize-318 TableCell-withFontSize-351"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-316 TableCell-withPadding-353 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-352 TableCell-withFontSize-318 TableCell-withFontSize-354"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-316 TableCell-withPadding-356 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-355 TableCell-withFontSize-318 TableCell-withFontSize-357"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root TableRow-withHover-343"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-316 TableCell-withPadding-359 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-358 TableCell-withFontSize-318 TableCell-withFontSize-360"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-316 TableCell-withPadding-362 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-361 TableCell-withFontSize-318 TableCell-withFontSize-363"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-316 TableCell-withPadding-365 TableCell-withVerticalAlign-315 TableCell-withVerticalAlign-364 TableCell-withFontSize-318 TableCell-withFontSize-366"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It adds sticky header 1`] = `
<div>
  <div
    class="MuiTableContainer-root TableContainer-withStickyHeader-369"
  >
    <table
      class="MuiTable-root MuiTable-stickyHeader"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-374"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-377 TableCell-withPadding-382 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-381 TableCell-withFontSize-379 TableCell-withFontSize-383 TableCell-withoutFirstLastPadding-378 MuiTableCell-stickyHeader"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-377 TableCell-withPadding-397 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-396 TableCell-withFontSize-379 TableCell-withFontSize-398 TableCell-withoutFirstLastPadding-378 MuiTableCell-stickyHeader"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-377 TableCell-withPadding-400 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-399 TableCell-withFontSize-379 TableCell-withFontSize-401 TableCell-withoutFirstLastPadding-378 MuiTableCell-stickyHeader"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-377 TableCell-withPadding-411 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-410 TableCell-withFontSize-379 TableCell-withFontSize-412 TableCell-withoutFirstLastPadding-378"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-377 TableCell-withPadding-414 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-413 TableCell-withFontSize-379 TableCell-withFontSize-415 TableCell-withoutFirstLastPadding-378"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-377 TableCell-withPadding-417 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-416 TableCell-withFontSize-379 TableCell-withFontSize-418 TableCell-withoutFirstLastPadding-378"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-377 TableCell-withPadding-420 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-419 TableCell-withFontSize-379 TableCell-withFontSize-421 TableCell-withoutFirstLastPadding-378"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-377 TableCell-withPadding-423 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-422 TableCell-withFontSize-379 TableCell-withFontSize-424 TableCell-withoutFirstLastPadding-378"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-377 TableCell-withPadding-426 TableCell-withVerticalAlign-376 TableCell-withVerticalAlign-425 TableCell-withFontSize-379 TableCell-withFontSize-427 TableCell-withoutFirstLastPadding-378"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It adds stripes 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-252"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-255 TableCell-withPadding-260 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-259 TableCell-withFontSize-257 TableCell-withFontSize-261"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-255 TableCell-withPadding-275 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-274 TableCell-withFontSize-257 TableCell-withFontSize-276"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-255 TableCell-withPadding-278 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-277 TableCell-withFontSize-257 TableCell-withFontSize-279"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root TableRow-withStripes-281"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-255 TableCell-withPadding-289 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-288 TableCell-withFontSize-257 TableCell-withFontSize-290"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-255 TableCell-withPadding-292 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-291 TableCell-withFontSize-257 TableCell-withFontSize-293"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-255 TableCell-withPadding-295 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-294 TableCell-withFontSize-257 TableCell-withFontSize-296"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root TableRow-withStripes-281"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-255 TableCell-withPadding-298 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-297 TableCell-withFontSize-257 TableCell-withFontSize-299"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-255 TableCell-withPadding-301 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-300 TableCell-withFontSize-257 TableCell-withFontSize-302"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-255 TableCell-withPadding-304 TableCell-withVerticalAlign-254 TableCell-withVerticalAlign-303 TableCell-withFontSize-257 TableCell-withFontSize-305"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It renders 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-8"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-11 TableCell-withPadding-16 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-15 TableCell-withFontSize-13 TableCell-withFontSize-17 TableCell-withoutFirstLastPadding-12"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-11 TableCell-withPadding-31 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-30 TableCell-withFontSize-13 TableCell-withFontSize-32 TableCell-withoutFirstLastPadding-12"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-11 TableCell-withPadding-34 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-33 TableCell-withFontSize-13 TableCell-withFontSize-35 TableCell-withoutFirstLastPadding-12"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-11 TableCell-withPadding-45 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-44 TableCell-withFontSize-13 TableCell-withFontSize-46 TableCell-withoutFirstLastPadding-12"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-11 TableCell-withPadding-48 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-47 TableCell-withFontSize-13 TableCell-withFontSize-49 TableCell-withoutFirstLastPadding-12"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-11 TableCell-withPadding-51 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-50 TableCell-withFontSize-13 TableCell-withFontSize-52 TableCell-withoutFirstLastPadding-12"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-11 TableCell-withPadding-54 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-53 TableCell-withFontSize-13 TableCell-withFontSize-55 TableCell-withoutFirstLastPadding-12"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-11 TableCell-withPadding-57 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-56 TableCell-withFontSize-13 TableCell-withFontSize-58 TableCell-withoutFirstLastPadding-12"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-11 TableCell-withPadding-60 TableCell-withVerticalAlign-10 TableCell-withVerticalAlign-59 TableCell-withFontSize-13 TableCell-withFontSize-61 TableCell-withoutFirstLastPadding-12"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It uses medium padding 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-130"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-133 TableCell-withPadding-138 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-137 TableCell-withFontSize-135 TableCell-withFontSize-139 TableCell-withoutFirstLastPadding-134"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-133 TableCell-withPadding-153 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-152 TableCell-withFontSize-135 TableCell-withFontSize-154 TableCell-withoutFirstLastPadding-134"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-133 TableCell-withPadding-156 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-155 TableCell-withFontSize-135 TableCell-withFontSize-157 TableCell-withoutFirstLastPadding-134"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-133 TableCell-withPadding-167 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-166 TableCell-withFontSize-135 TableCell-withFontSize-168 TableCell-withoutFirstLastPadding-134"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-133 TableCell-withPadding-170 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-169 TableCell-withFontSize-135 TableCell-withFontSize-171 TableCell-withoutFirstLastPadding-134"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-133 TableCell-withPadding-173 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-172 TableCell-withFontSize-135 TableCell-withFontSize-174 TableCell-withoutFirstLastPadding-134"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-133 TableCell-withPadding-176 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-175 TableCell-withFontSize-135 TableCell-withFontSize-177 TableCell-withoutFirstLastPadding-134"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-133 TableCell-withPadding-179 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-178 TableCell-withFontSize-135 TableCell-withFontSize-180 TableCell-withoutFirstLastPadding-134"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-133 TableCell-withPadding-182 TableCell-withVerticalAlign-132 TableCell-withVerticalAlign-181 TableCell-withFontSize-135 TableCell-withFontSize-183 TableCell-withoutFirstLastPadding-134"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It uses small padding 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <thead
        class="MuiTableHead-root TableHead-whiteBackground-69"
        data-id="table-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-72 TableCell-withPadding-77 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-76 TableCell-withFontSize-74 TableCell-withFontSize-78 TableCell-withoutFirstLastPadding-73"
          data-id="table-cell"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-72 TableCell-withPadding-92 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-91 TableCell-withFontSize-74 TableCell-withFontSize-93 TableCell-withoutFirstLastPadding-73"
          data-id="table-cell"
          scope="col"
        >
          Age
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head TableCell-withPadding-72 TableCell-withPadding-95 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-94 TableCell-withFontSize-74 TableCell-withFontSize-96 TableCell-withoutFirstLastPadding-73"
          data-id="table-cell"
          scope="col"
        >
          Role
        </th>
      </thead>
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-72 TableCell-withPadding-106 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-105 TableCell-withFontSize-74 TableCell-withFontSize-107 TableCell-withoutFirstLastPadding-73"
            data-id="table-cell"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-72 TableCell-withPadding-109 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-108 TableCell-withFontSize-74 TableCell-withFontSize-110 TableCell-withoutFirstLastPadding-73"
            data-id="table-cell"
          >
            32
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-72 TableCell-withPadding-112 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-111 TableCell-withFontSize-74 TableCell-withFontSize-113 TableCell-withoutFirstLastPadding-73"
            data-id="table-cell"
          >
            Software Engineer
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-72 TableCell-withPadding-115 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-114 TableCell-withFontSize-74 TableCell-withFontSize-116 TableCell-withoutFirstLastPadding-73"
            data-id="table-cell"
          >
            John
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-72 TableCell-withPadding-118 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-117 TableCell-withFontSize-74 TableCell-withFontSize-119 TableCell-withoutFirstLastPadding-73"
            data-id="table-cell"
          >
            43
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-72 TableCell-withPadding-121 TableCell-withVerticalAlign-71 TableCell-withVerticalAlign-120 TableCell-withFontSize-74 TableCell-withFontSize-122 TableCell-withoutFirstLastPadding-73"
            data-id="table-cell"
          >
            Business Analyst
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
